internal/handler/withdrawal_handler_test.go

```diff
@@ -0,0 +1,214 @@
+			req = req.WithContext(mockUserIDContext(tc.userID))
+			rr := httptest.NewRecorder()
+
+			// Call the handler
+			withdrawHandler.Withdraw(rr, req)
+
+			// Validate the status code
+			if rr.Code != tc.expectedStatus {
+				t.Errorf("expected status code %d, got %d", tc.expectedStatus, rr.Code)
+			}
+		})
+	}
+}
+
+func TestWithdrawHandler_GetWithdrawals(t *testing.T) {
+	mockBalanceService := &MockBalanceService{}
+	mockWithdrawService := &MockWithdrawService{}
+	mockOrderService := &MockOrderService{}
+
+	withdrawHandler := handler.NewWithdrawHandler(mockBalanceService, mockWithdrawService, mockOrderService)
+
+	tests := map[string]struct {
+		userID               uint
+		mockGetUserWithdraws func(userID uint) ([]model.UserWithdrawal, error)
+		expectedStatus       int
+		expectedResponse     []handler.WithdrawalResponse
+	}{
+		"Valid Withdrawals": {
+			userID: 1,
+			mockGetUserWithdraws: func(userID uint) ([]model.UserWithdrawal, error) {
+				return []model.UserWithdrawal{
+					{
+						Order:       "123456789",
+						Sum:         50,
+						ProcessedAt: time.Now(),
+					},
+				}, nil
+			},
+			expectedStatus: http.StatusOK,
+			expectedResponse: []handler.WithdrawalResponse{
+				{
+					Order:       "123456789",
+					Sum:         50,
+					ProcessedAt: time.Now().Format(time.RFC3339),
+				},
+			},
+		},
+		"No Withdrawals": {
+			userID: 1,
+			mockGetUserWithdraws: func(userID uint) ([]model.UserWithdrawal, error) {
+				return []model.UserWithdrawal{}, nil
+			},
+			expectedStatus:   http.StatusNoContent,
+			expectedResponse: nil,
+		},
+		"Internal Server Error": {
+			userID: 1,
+			mockGetUserWithdraws: func(userID uint) ([]model.UserWithdrawal, error) {
+				return nil, errors.New("internal error")
+			},
+			expectedStatus:   http.StatusInternalServerError,
+			expectedResponse: nil,
+		},
+	}
+
+	for name, tc := range tests {
+		t.Run(name, func(t *testing.T) {
+			// Mock withdraw service functions
+			mockWithdrawService.GetUserWithdrawalsFunc = tc.mockGetUserWithdraws
+
+			// Create a request
+			req := httptest.NewRequest("GET", "/withdrawals", nil)
+			req = req.WithContext(mockUserIDContext(tc.userID))
+			rr := httptest.NewRecorder()
+
+			// Call the handler
+			withdrawHandler.GetWithdrawals(rr, req)
+
+			// Validate the status code
+			if rr.Code != tc.expectedStatus {
+				t.Errorf("expected status code %d, got %d", tc.expectedStatus, rr.Code)
+			}
+
+			// Validate the response if applicable
+			if tc.expectedStatus == http.StatusOK {
+				var response []handler.WithdrawalResponse
+				err := json.NewDecoder(rr.Body).Decode(&response)
+				if err != nil {
+					t.Fatalf("failed to decode response: %v", err)
+				}
+
+				// Compare expected and actual response
+				for i, expected := range tc.expectedResponse {
+					if response[i].Order != expected.Order || response[i].Sum != expected.Sum {
+						t.Errorf("expected %v, got %v", expected, response[i])
+					}
+				}
```

Comment:
Вместо проверки значений в цикле можно просто сравнить два массива. Это удобно, так как это покажет все несоответствия в двух структурах.

```suggestion
                if diff := cmp.Diff(tc.expectedResponse, response); diff != "" {
			t.Errorf("Unexpected response (-expected +got):\n%s", diff)
		}
```